### DISCUSSION TITLE
I want to contribute a custom Oauth Provider

### DISCUSSION THREAD

**Discussion by Mutondi** [supabase/supabase#4704] [Questions] [answered]
Good Day, 

I would like to contribute a custom OAuth provider (Xero / Auth0) due to some business requirements. 

I am not too familiar with the codebase so I thought I will post here, can someone please direct me in terms of the files that I need to change to effect this change.

**Comment by J0** [comment]
Hey @Mutondi,

Thanks for the query, there isn't a detailed guide right now for adding another custom provider like "Twitter"  or "Facebook" but I believe there is a blog post in the works. For now, your best bet might be to refer to one of the existing PRs to provide Auth providers: https://github.com/supabase/gotrue/pull/269

For adding custom providers via config there is this thread: https://github.com/supabase/supabase/discussions/417 but it has yet to be updated.

**Comment by J0** [comment, marked as answer]
Hey,

For future readers, please feel free to posts provider requests on our [Provider requests thread](https://github.com/supabase/gotrue/issues/451)  and upvote providers which you'd like to see. We will provide guidance in terms of implementation for providers which are relevant and sought after by users.

Thanks!
